akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -1359,6 +1386,42 @@ final class Flow[In, Out, Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends Gr
   def recoverWith(pf: PartialFunction[Throwable, _ <: Graph[SourceShape[Out], NotUsed]]): javadsl.Flow[In, Out, Mat] =
     new Flow(delegate.recoverWith(pf))
 
+  /**
+    * RecoverWith allows to switch to alternative Source on flow failure. It will stay in effect after
+    * a failure has been recovered so that each time there is a failure it is fed into the `pf` and a new
+    * Source may be materialized.
+    *
+    * Since the underlying failure signal onError arrives out-of-band, it might jump over existing elements.
+    * This stage can recover the failure signal, but not the skipped elements, which will be dropped.
+    *
+    * Throwing an exception inside `recoverWith` _will_ be logged on ERROR level automatically.
+    *
+    * '''Emits when''' element is available from the upstream or upstream is failed and element is available
+    * from alternative Source
+    *
+    * '''Backpressures when''' downstream backpressures
+    *
+    * '''Completes when''' upstream completes or upstream failed with exception pf can handle
+    *
+    * '''Cancels when''' downstream cancels
+    *
+    */
+  @deprecated("Use recoverWithRetries instead.", "2.4.4")
```

Comment:
Shall we remove the deprecation of `recoverWith`? (question to Akka team)

Reply:
I think so. Never understood why it was deprecated to begin with.

Reply:
I will remove it later.

Reply:
Thanks!